db/migrate/20241216104132_add_deleted_column_in_course_user.rb

```diff
@@ -0,0 +1,6 @@
+class AddDeletedColumnInCourseUser < ActiveRecord::Migration[7.0]
+  def change
+    add_column :course_users, :deleted_at, :datetime
+    add_column :course_users, :deleter_id, :integer
```

Comment:
Clarification point: Are we planning to create FK to the user through `deleted_id`?

Reply:
I don't see any need for creating FK for this one, since eventually the course user will be deleted (should this columns ever be filled in) and thus, in most of the time, these 2 columns will be `nil`

Reply:
In that case `deleter_id` will also be ephemeral and isn't needed, we can remove it.

Reply:
removed